### ISSUE TITLE
Create a GitHub app for nixpkgs-vet releases

### ISSUE THREAD

**Issue by infinisil** [nix-open-org/org-v2#26]
Also [asked on Matrix](https://matrix.to/#/!DMULJFTtTWdmyAxeLd:nixos.org/$A5514665p_5Rh9Rvpzati8n183nuSI9kPjHj5N0RBD8?via=nixos.org&via=matrix.org&via=lassul.us):

I need a GitHub App to create releases for [nixpkgs-vet](https://github.com/NixOS/nixpkgs-vet) (formerly nixpkgs-check-by-name), see [here](https://github.com/NixOS/nixpkgs-vet/pull/97#issuecomment-2317942367) for more context.

Instructions:

- Create new App ([direct link](https://github.com/organizations/NixOS/settings/apps/new)) with permissions Repository->Content->Read/Write, name can be "Nixpkgs-vet releases"
- Install the App on the org, give it access to only https://github.com/nixos/nixpkgs-vet
- Set the App ID as a new _repository variable_ ([direct link](https://github.com/NixOS/nixpkgs-vet/settings/variables/actions/new)) named `RELEASES_APP_ID`
- Generate a private key for the App and set it as a new _repository secret_ ([direct link](https://github.com/NixOS/nixpkgs-vet/settings/secrets/actions/new)) named `RELEASES_PRIVATE_KEY`

I will then be able to use [this GitHub Action](https://github.com/actions/create-github-app-token) to get ephemeral GitHub App installation tokens for CI

**Comment by zimbatm** [comment]
@infinisil do you still need this?

**Comment by JohnRTitor** [comment]
Instead of having multiple apps for multiple things, I think it would be better to create a common app which will be used for multiple purposes. ie, for CI in Nixpkgs, nix-owners, nix-backports can be squashed into a single nixpkgs-ci / nix-ci bot

**Comment by roberth** [comment]
Might be better for security this way.
Should become easy to manage with #40 
Apps can be assigned users to manage them on a per-app basis. This may be useful when you're using actual GitHub Apps features and not just a token to feed back into GitHub Actions.

**Comment by infinisil** [comment]
@JohnRTitor Problem is that each app needs different permissions, and they shouldn't have more permissions than necessary:
- nix-owners: Needs Nixpkgs PR write permission
- nix-backports: Needs no permissions
- nixpkgs-vet: Needs nixpkgs-vet content write permission

**Comment by infinisil** [comment]
> @infinisil do you still need this?

Not urgently, it's a _very_ low priority thing, but it would be nice, not even just due to more automation, but to also try out this automation concept.

**Comment by winterqt** [comment]
> nix-backports: Needs no permissions

This needs Nixpkgs PR write and repo write.

**Comment by infinisil** [comment]
Oh right, I should know because I requested it in https://github.com/NixOS/org/issues/38 :laughing: 

We can probably indeed unify some use cases in the same bot

**Comment by wolfgangwalther** [comment]
> > nix-backports: Needs no permissions
> 
> This needs Nixpkgs PR write and repo write.

With #133, it will also need "issues: write". It would be good to *not* use this `nixpkgs-ci` app for more than.. nixpkgs-ci! That's because it has a rate limit of 12.5k req / hour and we're going somewhere in the range of 6-7k / hour right now to label all PRs. We'd like to implement more labeling - and it would be good if no other factors, outside of our control, contributed to that limit.

> nix-owners: Needs Nixpkgs PR write permission

Note, that these are actually two apps, AFAICT. One read-only, one read-write. Imho this is not necessary anymore today, because `actions/create-github-app-token` allows specifying only a subset of the available app privileges for the token to be created. Thus, the token for the codeowners/check can still be read-only, even if both use the same read/write app.

Since this is inside Nixpkgs and doesn't use many requests, it would make a lot of sense to move it entirely into the `nixpkgs-ci` app, though.

At the same time, an app for nixpkgs-vet should be separated. Also, it makes little sense to create one app to "update all kinds of repos across the NixOS org", because that would essentially allow privilege escalation across repositories. Somebody with repo-admin access to nixpkgs-vet could create a token that allows access to other repos managed by that app. I guess that's not what we want.

TLDR: I think we should probably go roughly towards "1 app per repo"?

**Comment by infinisil** [comment]
Not relevant anymore with https://github.com/NixOS/nixpkgs-vet/pull/244 :)
